pypeit/reduce.py

```diff
@@ -391,14 +389,19 @@ def run_objfind(self, std_trace=None, show_peaks=False):
         else:
             msgs.info("Skipping 2nd run of finding objects")
 
+        # ##################################################
+        # Second pass global sky
         # Do we have any positive objects to proceed with?
-        if self.nobj > 0:
-            # Global sky subtraction second pass. Uses skymask from object finding
-            if (self.std_redux or self.par['reduce']['extraction']['skip_optimal'] or
-                    self.par['reduce']['findobj']['skip_second_find'] or usersky):
-                self.global_sky = self.initial_sky.copy()
-            else:
-                self.global_sky = self.global_skysub(skymask=self.skymask, show=self.reduce_show)
+        #if self.nobj > 0:
+
+        # Global sky subtraction second pass. Uses skymask from object finding
```

Comment:
Why do you now require a second pass? If there are no objects identified, then the second pass is going to produce exactly the same result as the first pass.

Reply:
The reason for this PR is to update the variance on the second pass.

We should do it, even for a slit without a detected object (although the
fact that we don't mask any pixels will lead to an underestimate of the
variance, as we have seen).

In a separate PR, @deborape will eliminate this step for instruments 
where we insert targeted objects.